socialhome/streams/views.py

```diff
@@ -1,35 +1,28 @@
 from braces.views import LoginRequiredMixin
 from django.shortcuts import get_object_or_404
-from django.utils.translation import gettext
 from django.views.generic import ListView
-from django.urls import reverse
 
 from socialhome.content.models import Content, Tag
 
 
-class BaseStreamView(ListView):
-    model = Content
-    ordering = "-created"
-    paginate_by = 30
-    stream_name = ""
+class VueStreamMixin:
     vue = False
+    __request = None
 
-    def dispatch(self, request, *args, **kwargs):
+    def _dispatch(self, request):
         use_new_stream = (
             hasattr(request.user, "preferences") and request.user.preferences.get("streams__use_new_stream")
         )
         self.vue = bool(request.GET.get("vue", False)) or use_new_stream
-        return super().dispatch(request, *args, **kwargs)
+        self.__request = request
 
-    def get_context_data(self, **kwargs):
-        context = super().get_context_data(**kwargs)
-        context["stream_name"] = self.stream_name
+    def get_json_context(self, **kwargs):
         if self.vue:  # pragma: no cover
-            context["json_context"] = self._define_context_json(self.stream_name)
-        return context
+            return self._define_context_json(kwargs["stream_name"], kwargs["content_list"])
```

Comment:
I don't see why we need to check `if self.vue` here - just return the JSON context. Rather, make sure this is not called if not `self.vue`.